lib/B/Deparse.pm

```diff
@@ -1180,6 +1181,94 @@ sub pad_subs {
 }
 
 
+# deparse_multiparam(): deparse, if possible, a sequence of ops into a
+# subroutine signature. If possible, returns a string representing the
+# signature syntax, minus the surrounding parentheses.
+
+sub deparse_multiparam {
+    my ($self, $topop, $cv) = @_;
+
+    $topop = $topop->first;
+    return unless $$topop and $topop->name eq 'lineseq';
+
+    # last op should be nextstate
+    my $last = $topop->last;
+    return unless $$last
+                    and (   _op_is_or_was($last, OP_NEXTSTATE)
+                         or _op_is_or_was($last, OP_DBSTATE));
+
+    # first OP_NEXTSTATE
+
+    my $o = $topop->first;
+    return unless $$o;
+    return if $o->label;
+
+    # OP_MULTIPARAM
+
+    $o = $o->sibling;
+    return unless $$o and $o->name eq 'multiparam';
+
+    my ($min_args, $max_args, $slurpy, @rest) = $o->aux_list($cv);
+    my $nparams = $max_args;
+    my @param_padix = splice @rest, 0, $nparams, ();
+    my ($slurpy_padix) = @rest;
+
+    my @sig;
+    my %parami_for_padix;
+
+    # Initial scalars
+    foreach my $parami ( 0 .. $max_args-1 ) {
+        my $padix = $param_padix[$parami];
+        $sig[$parami] = $self->padname($padix) || '$';
+        $parami_for_padix{$padix} = $parami;
+    }
+
+    $o = $o->sibling;
+    for (; $o and !null $o; $o = $o->sibling) {
+        # Look for OP_NULL[OP_PARAMTEST[OP_PARAMSTORE]]
+        if ($o->name eq 'null' and $o->flags & OPf_KIDS and
+                $o->first->name eq 'paramtest' and
+                $o->first->first->name eq 'paramstore') {
+            # A defaulting expression
+            my $paramtest = $o->first;
+
+            my $parami = $parami_for_padix{$paramtest->targ};
+
+            my $assign = "=";
+            $assign = "//=" if $paramtest->private == OPpPARAM_IF_UNDEF;
+            $assign = "||=" if $paramtest->private == OPpPARAM_IF_FALSE;
+
+            length $sig[$parami] > 1 ?
+                ( $sig[$parami] .= ' ' ) :
+                ( $sig[$parami] = '$' ); # intentionally no trailing space
```

Comment:
dont look up same thing over and over in an array, do a `$sref = \$sig[$parami];` or `$s = $sig[$parami];`. probably the first is easier in this sub.